### ISSUE TITLE
ReactiveCommand.CanExecute can return false when no canExecute specified

### ISSUE THREAD

**Issue by csmager** [reactiveui/ReactiveUI#3040]
**Describe the bug**
Following the change in #2835 to add a scheduler for `canExecuteObservable` and default it to `RxApp.MainThreadScheduler`, we've now changed the behaviour - the result of `canExecute` will not be available until the scheduler runs.

**Steps To Reproduce**

This code now returns `false` where it used to return `true` - I assume because the default `Observables.True` is now scheduled and not yet run:
```csharp
ICommand command = ReactiveCommand.Create(() => { });
var canExecute = command.CanExecute(null);
```
This caused a bug in our software where we return some `ICommands` to a 3rd party control to execute on a key-press. The control calls `CanExecute` first - the commands that used to execute on key-press are now not.

**Expected behaviour**

I'd expect that if no `canExecute` observable is provided that the scheduler used would be `Scheduler.Immediate` given the value will never change. This is my current workaround.

**Comment by anaisbetts** [comment]
@glennawatson  I think that canExecute should not be implicitly scheduled at all, it should be Expected that it is already coming from the UI thread (because in the vast majority of cases it is already guaranteed to be on the UI thread)

**Comment by glennawatson** [comment]
Had a talk to @ChrisPulman about this one. We will revert the change introduced and then create better documentation for users who use schedulers other than the `MainThreadScheduler` to use a `ObserveOn()`

Leaving this issue open to track those series of changes.

**Comment by glennawatson** [comment]
`WhenAny` isn't guaranteed last time I checked btw, will be whatever thread runs the set unless observed elsewhere. This should be documented if it's not already anyway as a possible interaction here with RxCommand

**Comment by glennawatson** [comment]
This was fixed with [#2863 ](https://github.com/reactiveui/ReactiveUI/pull/3045)
